tests/subsys/usb_c/sink/src/main.c

```diff
@@ -0,0 +1,514 @@
+
+	data->pn_protocol_error = ATOMIC_INIT(false);
+	data->pn_msg_discarded = ATOMIC_INIT(false);
+	data->pn_msg_accept_received = ATOMIC_INIT(false);
+	data->pn_msg_rejected_received = ATOMIC_INIT(false);
+	data->pn_msg_not_supported_received = ATOMIC_INIT(false);
+	data->pn_transition_ps = ATOMIC_INIT(false);
+	data->pn_pd_connected = ATOMIC_INIT(false);
+	data->pn_not_pd_connected = ATOMIC_INIT(false);
+	data->pn_power_change_0a0 = ATOMIC_INIT(false);
+	data->pn_power_change_def = ATOMIC_INIT(false);
+	data->pn_power_change_1a5 = ATOMIC_INIT(false);
+	data->pn_power_change_3a0 = ATOMIC_INIT(false);
+	data->pn_data_role_is_ufp = ATOMIC_INIT(false);
+	data->pn_data_role_is_dfp = ATOMIC_INIT(false);
+	data->pn_port_partner_not_responsive = ATOMIC_INIT(false);
+	data->pn_snk_transition_to_default = ATOMIC_INIT(false);
+	data->pn_hard_reset_received = ATOMIC_INIT(false);
+	data->pn_source_capabilities_received = ATOMIC_INIT(false);
+	data->pn_sender_response_timeout = ATOMIC_INIT(false);
+
+	data->uut_received_src_cap_num = 0;
+	data->snk_cap_pdo.raw_value = 0;
+	data->rx_msg.len = 0;
+
+	/* Initialize PDO for sending in step 5 */
+
+	/* a) B31…30 (Fixed Supply) set to 00b */
+	pdo.type = PDO_FIXED;
+	/* b) B29 (Dual-Role Power) set to 1b */
+	pdo.dual_role_power = 1;
+	/* c) B28 (USB Suspend Supported) set to 0b */
+	pdo.usb_suspend_supported = 0;
+	/* d) B27 (Unconstrained Power) set to 1b */
+	pdo.unconstrained_power = 1;
+	/* e) B26 (USB Communications Capable) set to 0b */
+	pdo.usb_comms_capable = 0;
+	/* f) B25 (Dual-Role Data) set to 0b */
+	pdo.dual_role_data = 0;
+	/* g) B24 (PD3, Unchunked Extended Messages Supported) set to 0b */
+	pdo.unchunked_ext_msg_supported = 0;
+	/*
+	 * h) B23 (EPR Mode Capable) to 0b, unless it is mentioned in the
+	 *    test procedure. NOTE: NOT CURRENTLY SUPPORTED IN THE SUBSYSTEM.
+	 */
+	pdo.reserved0 = 0;
+	/* i) B21…20 (Peak Current) set to 00b */
+	pdo.peak_current = 0;
+	/* j) B19…10 (Voltage) set to 5V */
+	pdo.voltage = PD_CONVERT_MV_TO_FIXED_PDO_VOLTAGE(5000);
+	/* k) B9…0 (Maximum Current) set to 100mA */
+	pdo.max_current = PD_CONVERT_MA_TO_FIXED_PDO_CURRENT(100);
+
+	/* 1) The test starts in a disconnected state. */
+	tester_disconnected();
+	/* Give the Sink state machine time to transition */
+	k_msleep(20);
+
+	/* 2) Apply Rp */
+	tester_apply_cc(TC_CC_VOLT_RP_3A0, TC_CC_VOLT_OPEN);
+
+	/* 3) Apply vSafe5V on VBUS. */
+	tester_apply_vbus(PD_V_SAFE_5V_MIN_MV);
+
+	/* 4) The Tester waits until TC_ATTACHED_SNK state is reached. */
+	for (i = 0; i < 500; i++) {
+		/* NOT_PD_CONNECTED notification is sent when the PE starts up */
+		if (atomic_get(&data->pn_not_pd_connected)) {
+			break;
+		}
+		k_usleep(500);
+	}
+
+	/* 5) Send up to 50 source capability messages to UUT */
+	for (i = 0, j = 0; i < 50; i++) {
+		switch (j) {
+		case 0:
+			/* Send Source Cap message */
+			tester_send_data_msg(PD_DATA_SOURCE_CAP, &pdo.raw_value, 1, true);
+			j++;
+			k = 0;
```

Comment:
All of these single-character counters are a bit hard to follow. Can you unnest this a bit or comment more about the flow?